packages/predicate/src/predicate.ts

```diff
@@ -27,6 +27,7 @@ export class Predicate<ARGS extends InputValue[]> extends Account {
 
   constructor(
     bytes: BytesLike,
+    chainId: number,
```

Comment:
Since we need the `chainId` to calculate the predicate's 'address', we need users to pass it in when initializing a predicate.